pkg/service/manipulator.go

```diff
@@ -172,6 +189,9 @@ func trackDuration(name string, start time.Time, spec processSpec) *metrics.Upda
 }
 
 // NewManipulator takes in a Processor interface and returns a new Manipulator
-func NewManipulator(processor processor.Processor) Manipulator {
-	return &manipulator{processor: processor}
+func NewManipulator(processor processor.Processor, defaultParams map[string]string) Manipulator {
```

Comment:
Can we make this a pointer to a map? So that passing nil is easier in case no defaults are needed?

Reply:
yeah, makes sense. i'll update this.

Reply:
@ajatprabha actually map is a reference type, so it can accept nil. i am just going to update the tests to pas nil instead of `map[string]string`.